frame/system/src/lib.rs

```diff
@@ -1147,13 +1148,14 @@ impl<T: Trait> Module<T> {
 	}
 
 	/// To be called immediately after an extrinsic has been applied.
-	pub fn note_applied_extrinsic(r: &DispatchOutcome, _encoded_len: u32, info: DispatchInfo) {
+	pub fn note_applied_extrinsic(r: &DispatchResultWithPostInfo, mut info: DispatchInfo) {
+		info.weight = extract_actual_weight(r, &info);
```

Comment:
I think this could get confusing?
I expect objects of type `DispatchInfo` to contain the information of the dispatch before being executed.
If we want the `DispatchInfo` / `PostDispatchInfo` distinction we should also keep to it?

Reply:
We want to keep the amount of data inside events to a minimum as they increase chain size (they are stored in their blocks). I do not think that the pre dispatch weight is of interest any longer when you know the correct weight.

@bkchr @gavofyork What do you think?

Reply:
I'm not so much arguing for keeping the `PreDispatchInfo` or weight around.
I would argue more for "if this is post dispatch, let's return/store the `PostDispatchInfo` (and add all the fields we think are necessary to it)"

Reply:
I just want to keep the semantics of `DispatchInfo` and `PostDispatchInfo` cleaner to make it easier to comprehend.

Reply:
I understand. But I don't think that this justifies including redundant information. So if you do not override the `weight` in `DispatchInfo` the event would look like this:

```
struct ExtrinsicInfo {
        pub class: DispatchClass,
	pub pays_fee: Pays
};
decl_event! {
    ExtrinsicSuccess((ExtrinsicInfo, PostDispatchInfo))
};
```

So we are basically duplicating the `DispatchInfo` as `ExtrinsicInfo` just in order to remove one field. I have the feeling that this does not improve things.

Reply:
Why not add it to the `PostDispatchInfo`?

Reply:
`class` and `pays_fee`? Those are no *post* infos. Those are *pre*  info. It makes no sense to return them from every `Dispatchable`.

Reply:
How about taking a middle gourd and making a _new struct_ that contains _only_ the stuff that we need inside the event and implementing a handy `From<(PostDispatchInfo, DispatchInfo)>` for it (or anything that makes it work easily).

Reply:
If we go this route, `DispatchInfo` is already not correct for the normal case. As `DispatchInfo` is acutally `PreDispatchInfo` that is generated "statically" and together with `PostDispatchInfo` we can generate the final `DispatchInfo`.

I'm fine with staying with `DispatchInfo` here and may introducing a `PreDispatchInfo` in the future.

Reply:
Yes I think this is where the original confusion stems from. `DispatchInfo` should be renamed to `PreDispatchInfo` and here we use the "new" `DispatchInfo` that only contains what we need here. Is any real opposition to keep it the way it is and to the rename in a follow up PR? Basically what @bkchr said.

Reply:
Sounds good to me.